pkg/types/baremetal/validation/platform.go

```diff
@@ -243,6 +243,23 @@ func validateHostsBMCOnly(hosts []*baremetal.Host, fldPath *field.Path) field.Er
 	})
 }
 
+// validateHostsBootMACAddress validates all hosts BootMACAddress are valid or not
+func validateHostsBootMACAddress(hosts []*baremetal.Host, fldPath *field.Path) field.ErrorList {
+	hostErrs := field.ErrorList{}
+
+	for idx, host := range hosts {
+		if host.BootMACAddress == "" {
+			continue
+		}
+
+		if err := validate.MAC(host.BootMACAddress); err != nil {
```

Comment:
Could be worth trying the already baked-in validator for mac? https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Media_Access_Control_Address_MAC

Reply:
Yes, `validate.MAC` function just wraps up ^ function https://github.com/openshift/installer/blob/3d19350885d593ee2b1d9ecd7612c2d697dab2a3/pkg/validate/validate.go#L231.

Reply:
I think it could be enough to just add `mac` here: https://github.com/openshift/installer/blob/969378dc498336d1ac29baa89f75324a44467d5d/pkg/types/baremetal/platform.go#L39?

Reply:
If it works, that would be much more simpler. I'll try.